lib/result.luau

```diff
@@ -106,44 +108,18 @@ end
 	@param err E
 	@return Result<T, E>
 ]=]
-function Result.new<T, E>(val: T, err: E)
+function Result.new<T, E>(val: T, err: E): Result<T, E>
+	local typeId: "Result" = "Result"
+	local self = {
+		typeId = typeId,
+	}
+	self._value = val
+	self._error = err
```

Comment:
Inline `self` as the first argument to `setmetatable` and cast the argument to `ResultInner` / `ResultData`.